### PR TITLE
Upgrade Checkstyle version

### PR DESCRIPTION
**Pull request by jenoudet**
### What changes are proposed in this pull request?
This PR upgrades the [maven checkstyle plugin](http://maven.apache.org/plugins/maven-checkstyle-plugin/) version to the latest version. This allows us to use [checkstyle semantics](https://checkstyle.sourceforge.io/) v9.0.1 to interface with the [IntelliJ checkstyle plugin](https://plugins.jetbrains.com/plugin/1065-checkstyle-idea)

The import order remains the same (static imports -> `alluxio.` imports -> all other imports -> `java.` and `javax.` imports). Each of these groups is separated by a newline. Imports are sorted in alphabetical order within each group. The previous version of checkstyle did not enforce any of these rules consistently. The new version of checkstyle does. The effect is that some imports have changed location within their groups (as they were not ordered alphabetically before), and some newlines within groups have been erased (the biggest culprit was an often found empty line between `java.` and `javax.` imports).

This change enables integration with the IntelliJ checkstyle plugin (which dropped support for the version of checkstyle we used before this change). This plugin allows the checkstyle checks to happen within the IDE, and can report checkstyle errors before committing to the repo, and without running `mvn checkstyle:checkstyle` on the command line.

### CODE REVIEW

**Comment by jenoudet** [comment]
@yuzhu @apc999 ptal, this PR upgrades the version of checkstyle we're using to better integrate with IntelliJ.

File: build/checkstyle/suppressions.xml [hunk 1]
```diff
@@ -8,4 +8,6 @@
     <!-- Suppresses files in the src/main folder. To be used for rules which only apply to test
          code. -->
     <suppress files="[\\/]src[\\/]main[\\/].*|[\\/]generated-sources[\\/].*" id="TestScope" />
+    <!--  Suppress specific file  -->
+    <suppress files="hub/server/src/main/java/alluxio/hub/manager/process/ManagerProcessContext.java" checks="Indentation" />
```
Comment (Xenorith) [inline build/checkstyle/suppressions.xml:12]: why is this file special?

**Comment by yuzhu** [comment]
@jenoudet did we change the ordering of the imports?? what was the reason behind that change? 
what would  not have been possible without these changes of import order?

File: pom.xml [hunk 2]
```diff
@@ -1205,6 +1200,14 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>3.1.2</version>
+        <dependencies>
+          <dependency>
+            <groupId>com.puppycrawl.tools</groupId>
+            <artifactId>checkstyle</artifactId>
+            <version>9.0.1</version>
```
Comment (Xenorith) [inline pom.xml:1208]: is this like a submodule of checkstyle? wondering why it's newly added

**Comment by jenoudet** [comment]
> @jenoudet did we change the ordering of the imports?? what was the reason behind that change? what would not have been possible without these changes of import order?

The import order remains the same (static imports -> `alluxio.` imports -> all other imports -> `java.` and `javax.` imports). Each of these groups is separated by a newline. Imports are sorted in alphabetical order within each group. The previous version of checkstyle did not enforce any of these rules consistently. The new version of checkstyle does. The effect is that some imports have changed location within their groups (as they were not ordered alphabetically before), and some newlines within groups have been erased (the biggest culprit was an often found empty line between `java.` and `javax.` imports).

This change enables integration with the IntelliJ checkstyle plugin (which dropped support for the version of checkstyle we used before this change). This plugin allows the checkstyle checks to happen within the IDE, and can report checkstyle errors before committing to the repo, and without running `mvn checkstyle:checkstyle` on the command line.

**Review by Xenorith — COMMENTED** [review]
i actually opened every file in the diff. it was an exercise in mouse clicking/carpal tunnel training

File: build/checkstyle/suppressions.xml (hunk 1 above)
Comment (jenoudet) [reply build/checkstyle/suppressions.xml:12]: It has a ton of indent violations which I did have the heart to deal with 1 by 1. 😞

File: pom.xml (hunk 2 above)
Comment (jenoudet) [reply pom.xml:1208]: 9.0.1 is the version of the [checkstyle language](https://checkstyle.sourceforge.io/releasenotes.html), which is independent from the version of the [maven checkstyle plugin](http://maven.apache.org/plugins/maven-checkstyle-plugin/history.html) 3.1.2

File: build/checkstyle/suppressions.xml (hunk 1 above)
Comment (Xenorith) [reply build/checkstyle/suppressions.xml:12]: ok we can leave this as an example and fix it later

**Comment by yuzhu** [comment]
> tic imports -> `alluxio.` imports -> all other imports -> `java.` and `javax.` imports). Each of these groups is separated by a newline. Imports are sorted in alphabetical order within each group. The previous version of checkstyle did not enforce any of these rules consistently. The new version of checkstyle does. The effect is that some imports have changed location within their groups (as they were not ordered alphabetically before), and some newlines within groups have been erased (the biggest culprit was an often found empty line between `java.` and `javax.` imports).

thanks for the detailed description. Now it makes sense to me. Maybe some of these can go into the PR description as well?

**Review by Xenorith — APPROVED** [review]

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by qian0817** [comment]
From the picture about intellij idea import order https://docs.alluxio.io/os/user/edge/en/contributor/Code-Conventions.html#coding-style. I'm confused whether I need to add a blank line between java and javax. we need to modify the docs here or the checkstyle rules need to be changed.
@jenoudet @Xenorith

**Comment by Xenorith** [comment]
@qian0817 thanks for pointing this out. @jenoudet let's update the docs too and make sure we are conveying consistent information

**Pull request merged**